src/freertos_drivers/common/WiFiInterface.hxx

```diff
@@ -0,0 +1,250 @@
+    ///
+    /// @param index index within saved profile list to remove, a value of 0xFF
+    /// will remove all stored profiles.
+    /// @return 0 upon success, else -1 on error.
+    int del_ssid_profile(const int index);
+
+    /// Retrieves a saved SSID profile by index.
+    ///
+    /// @param index of the stored profile to retrieve.
+    /// @param ssid is a 33 byte array containing the ssid of the profile.
+    /// @param sec_type is the security type of the profile.
+    /// @param priority is the priority of the profile.
+    /// @return 0 upon success, else -1 on error.
+    int get_ssid_profile(const int index, const std::string &ssid,
+                         const WiFiSecurity &sec_type,
+                         const uint8_t &priority);
+
+    /// @return true if there are no saved SSID profiles, false otherwise.
+    bool are_ssid_profiles_empty();
+
+    /// Retrieves a list of available access points.
+    ///
+    /// @param entries will contain a list of available network entries.
+    /// @param count is the number of network entries to retrieve, max of 20.
+    /// @return number of network entries retrieved.
+    int get_network_list(NetworkListEntry *entries, size_t count);
+
+    /// Initiate scanning for available access points.
+    ///
+    /// @param blocking will cause this method to block until the network scan
+    /// completes.
+    /// @return the number of networks found via scan, this value should be
+    /// ignored when blocking is false (default).
+    int network_list_scan(bool blocking = false);
+
+    /// Retrieves the WiFi interface MAC address.
+    ///
+    /// @param mac is a 6 byte array which will hold the resulting MAC address.
+    /// @param mode is the WiFi interface to retrieve the MAC address for, if
+    /// an interface does not support the passed mode it will set the mac to
+    /// all zeros.
+    /// @return zero for succesful retrieval of the MAC address, non-zero for
+    /// failure (ie: mode not supported).
+    int get_mac(uint8_t *mac, WiFiMode mode = WiFiMode::STATION);
+
+    /// Returns the IPv4 address assigned to the WiFi iterface.
+    ///
+    /// Note: This method can return zero if the passed mode is not supported
+    /// or the WiFi interface does not have an IPv4 address for a supported
+    /// WiFi WiFiMode.
+    ///
+    /// @param mode is the WiFi interface to return the IPv4 address for.
+    /// @return the IPv4 address for the WiFi interface for the mode provided.
+    uint32_t get_ip(WiFiMode mode = WiFiMode::STATION);
+
+    /// @return SSID of the access point we are connected to, can return
+    /// nullptr if not currently connected to an access point. When the
+    /// interface is operating as an ACCESS POINT only this can return the 
+    /// SSID of the ACCESS POINT operating on the WiFi interface.
+    const std::string get_ssid();
+
+    /// @return RSSI value of the currently connected access point, can return
```

Comment:
(usually a negative number in dB)

Reply:
I'd expect it always to be negative values. Perhaps zero for not connected and positive for not supported/known?